tests/unit/states/test_saltutil.py

```diff
@@ -45,6 +48,8 @@ def test_saltutil_sync_all_nochange(self):
             'proxymodules': [],
             'output': [],
             'pillar': [],
+            'matchers': [],
+            'serializers': [],
```

Comment:
@mchugh19 Could you please clarify when these tests are supposed to fail? For me they pass even if I remove these two lines (`matchers` and `serializers`).

Reply:
@mchugh19 ping.

Reply:
Just took a look, and it appears that as long as there are no changes to sync, the test succeeds. So for example, if pillar had data: `'pillar': ['somedata'],` that causes the test to fail (since state would interpret that there are changes to sync, when it should return no changes.)

Thus, as of now, the test is only validating that the state evaluates empty sync types as not having any changes. It does not otherwise understand what types *should* be synced. I'm not sure if having that logic included is useful or not. In its current form, the sync state walks the module output looking for changes so it's really only the sync modules that need to be updated with additional types.